docs/howtos/convert_pytorch_to_flax.rst

```diff
@@ -0,0 +1,351 @@
+      strides: a sequence of `n` integers, representing the inter-window
+          strides.
+      padding: either the string `'SAME'`, the string `'VALID'`, or a sequence
+        of `n` `(low, high)` integer pairs that give the padding to apply before
+        and after each spatial dimension.
+    Returns:
+      The output of the reduction for each window slice.
+    """
+    strides = strides or (1,) * len(window_shape)
+    assert len(window_shape) == len(strides), (
+        f"len({window_shape}) == len({strides})")
+    strides = (1,) + strides + (1,)
+    dims = (1,) + window_shape + (1,)
+  
+    is_single_input = False
+    if inputs.ndim == len(dims) - 1:
+      # add singleton batch dimension because lax.reduce_window always
+      # needs a batch dimension.
+      inputs = inputs[None]
+      is_single_input = True
+  
+    assert inputs.ndim == len(dims), f"len({inputs.shape}) != len({dims})"
+    if not isinstance(padding, str):
+      padding = tuple(map(tuple, padding))
+      assert(len(padding) == len(window_shape)), (
+        f"padding {padding} must specify pads for same number of dims as "
+        f"window_shape {window_shape}")
+      assert(all([len(x) == 2 for x in padding])), (
+        f"each entry in padding {padding} must be length 2")
+      padding = ((0,0),) + padding + ((0,0),)
+    y = jax.lax.reduce_window(inputs, init, reduce_fn, dims, strides, padding)
+    if is_single_input:
+      y = jnp.squeeze(y, axis=0)
+    return y
+ 
+
+The above function is taken from |nn.pooling|_ and is the core function behind |nn.avg_pool()|_ and |nn.max_pool()|_.
+We then simply wrap this function with our own ``avg_pool()`` function.
+
+.. |nn.avg_pool()| replace:: ``nn.avg_pool()``
+.. _nn.avg_pool(): https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.avg_pool.html
+
+.. |nn.max_pool()| replace:: ``nn.max_pool()``
+.. _nn.max_pool(): https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.max_pool.html
+
+.. |nn.pooling| replace:: ``nn.pooling``
+.. _nn.pooling: https://github.com/google/flax/blob/main/flax/linen/pooling.py
+
+.. testcode::
+  
+  def avg_pool(inputs, window_shape, strides=None, padding='VALID'):
+    """
+    Pools the input by taking the average over a window.
+    In comparison to nn.avg_pool(), this pooling operation does not
+    consider the padded zero's for the average computation.
+    Args:
+      inputs: input data with dimensions (batch, window dims..., features).
+      window_shape: a shape tuple defining the window to reduce over.
+      strides: a sequence of `n` integers, representing the inter-window
+          strides (default: `(1, ..., 1)`).
+      padding: either the string `'SAME'`, the string `'VALID'`, or a sequence
+        of `n` `(low, high)` integer pairs that give the padding to apply before
+        and after each spatial dimension (default: `'VALID'`).
+    Returns:
+      The average for each window slice.
+    """
+    assert inputs.ndim == 4
```

Comment:
nit: I think the updated implementation should work with arbitrary `ndim` like the `nn.avg_pool()`